### PR TITLE
⛳️  Golf the LLVM CMAKE options in the Dockerfiles.

### PR DESCRIPTION
**Pull request by Choc13**

### CODE REVIEW

File: lib/build/Dockerfile [hunk 1]
```diff
@@ -1,29 +1,27 @@
...
-        -DLLVM_ENABLE_PROJECTS=clang \
         -DCMAKE_BUILD_TYPE=MinSizeRel \
+        -DLLVM_BUILD_TOOLS=OFF \
+        -DLLVM_ENABLE_PROJECTS=clang \
+        -DLLVM_INCLUDE_EXAMPLES=OFF \
+        -DLLVM_INCLUDE_TESTS=OFF \
+        -DLLVM_PARALLEL_LINK_JOBS=${llvmParallelJobs} \
         -DLLVM_TARGETS_TO_BUILD=Native \
-        -DLLVM_PARALLEL_LINK_JOBS=${clangParallelJobs} \
-        -DLLVM_ENABLE_EH=ON \
```
Comment (Choc13) [inline lib/build/Dockerfile:20]: According to the docs:

> Build LLVM with exception-handling support. This is necessary if you wish to link against LLVM libraries and make use of C++ exceptions in your own code that need to propagate through LLVM code. Defaults to OFF.

I don't believe we need this feature.

File: lib/build/Dockerfile [hunk 2]
```diff
@@ -1,29 +1,27 @@
...
-        -DLLVM_ENABLE_FFI=NO \
```
Comment (Choc13) [inline lib/build/Dockerfile:21]: This defaults to `OFF` anyway.

File: lib/build/Dockerfile [hunk 3]
```diff
@@ -1,29 +1,27 @@
...
-        -DLLVM_ENABLE_LIBCXX=NO \
```
Comment (Choc13) [inline lib/build/Dockerfile:22]: This defaults to `OFF` anyway.

File: lib/build/Dockerfile [hunk 4]
```diff
@@ -1,29 +1,27 @@
...
-        -DLLVM_ENABLE_PIC=YES \
```
Comment (Choc13) [inline lib/build/Dockerfile:23]: This defaults to `ON` anyway.

File: lib/build/Dockerfile [hunk 5]
```diff
@@ -1,29 +1,27 @@
...
-        -DLLVM_ENABLE_RTTI=YES \
```
Comment (Choc13) [inline lib/build/Dockerfile:24]: From the docs:

> Build LLVM with run-time type information. Defaults to OFF.

My understanding is that this is a C++ feature and I don't think we need it / use it currently.

**Comment by Choc13** [comment]
This seems to have dropped 7MB off of the image size too 🥳

**Review by iscottb122 — APPROVED** [review]

**Pull request merged**